testutils/infra/container_manager/kubernetes_manager.py

```diff
@@ -42,22 +63,74 @@ def cmd(self, container_id, docker_cmd, cmd=[]):
         )
         return ret.stdout.decode("utf-8").strip()
 
-    def getid(self, filters):
-        filters = ["grep {}".format(f) for f in filters]
-        cmd = (
-            "kubectl get pods | "
-            + " | ".join(filters)
-            + " | awk '{print $1}' | head -n 1"
-        )
-        ret = subprocess.check_output(cmd, shell=True).decode("utf-8").strip()
-        if ret == "":
-            raise RuntimeError("container id for {} not found".format(str(filters)))
-        return ret
+    def getid(self, filters: List[str]) -> str:
+        """Get pod id based on given filters."""
+        pod_list = self.v1.list_namespaced_pod(self.namespace)
+        pod_id = ""
+
+        for pod in pod_list.items:
+            if filters[0] in pod.metadata.name and pod.status.phase == "Running":
+                pod_id = pod.metadata.name
+        if pod_id == "":
+            raise RuntimeError(f"pod id for filters {str(filters)} not found")
+        return pod_id
 
     def get_mender_gateway(self):
         return os.environ.get("GATEWAY_HOSTNAME")
 
 
+class KubernetesEnterpriseSetupWithGateway(KubernetesNamespace):
+    COMPOSE_FILES_PATH = os.path.realpath(
+        os.path.join(os.path.dirname(__file__), "..", "..", "..")
+    )
+    MENDER_GATEWAY_FILES = [
```

Comment:
Q: These are duplicated from `docker_compose_manager.py`, right?

Maybe we should split them out in both places, to make this consistent?

Reply:
Thanks for comment! I'll work on this in future PRs; let's see how it behaves on staging environment.